scripts/generateRules.mjs

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/env node
```

Comment:
I'm happy to land and then iteratie on this, but I would love to run this as a build step instead of committing the generated JSON.  Any objections?